src/snapred/backend/data/LocalDataService.py

```diff
@@ -303,230 +253,85 @@ def _constructReductionDataRoot(self, runNumber: str, useLiteMode: bool) -> Path
-        expressions = {
-            ">=": lambda x, y: x >= y,
-            "<=": lambda x, y: x <= y,
-            "<": lambda x, y: x < y,
-            ">": lambda x, y: x > y,
-            "": lambda x, y: x == y,
-        }
-        return expressions[symbol](int(runNumber1), int(runNumber2))
-
-    def _isApplicableEntry(self, calibrationIndexEntry, runId):
-        """
-        Checks to see if an entry in the calibration index applies to a given run id via numerical comparison.
-        """
+    def normalizationIndexor(self, runId: str, useLiteMode: bool):
+        return self.indexor(runId, useLiteMode, IndexorType.NORMALIZATION)
 
-        symbol, runNumber = self._parseAppliesTo(calibrationIndexEntry.appliesTo)
-        return self._compareRunNumbers(runId, runNumber, symbol)
+    def reductionIndexor(self, runId: str, useLiteMode: bool):
+        return self.indexor(runId, useLiteMode, IndexorType.REDUCTION)
 
-    def _getVersionFromCalibrationIndex(self, runId: str, useLiteMode: bool) -> int:
-        """
-        Loads calibration index and inspects all entries to attain latest calibration version that applies to the run id
-        """
-        # lookup calibration index
-        calibrationIndex = self.readCalibrationIndex(runId, useLiteMode)
-        # From the index find the latest calibration
-        latestCalibration = None
-        version = None
-        if calibrationIndex:
-            # sort by timestamp
-            calibrationIndex.sort(key=lambda x: x.timestamp)
-            # filter for latest applicable
-            relevantEntries = list(filter(lambda x: self._isApplicableEntry(x, runId), calibrationIndex))
-            if len(relevantEntries) < 1:
-                return None
-            latestCalibration = relevantEntries[-1]
-            version = latestCalibration.version
-        return version
-
-    def _getVersionFromNormalizationIndex(self, runId: str, useLiteMode: bool) -> int:
-        """
-        Loads normalization index and inspects all entries to attain
-        latest normalization version that applies to the run id
-        """
-        # lookup normalization index
-        normalizationIndex = self.readNormalizationIndex(runId, useLiteMode)
-        # From the index find the latest normalization
-        latestNormalization = None
-        version = None
-        if normalizationIndex:
-            # sort by timestamp
-            normalizationIndex.sort(key=lambda x: x.timestamp)
-            # filter for latest applicable
-            relevantEntries = list(filter(lambda x: self._isApplicableEntry(x, runId), normalizationIndex))
-            if len(relevantEntries) < 1:
-                return None
-            latestNormalization = relevantEntries[-1]
-            version = latestNormalization.version
-        return version
-
-    def getVersionFromNormalizationIndex(self, runId: str, useLiteMode: bool) -> int:
-        return self._getVersionFromNormalizationIndex(runId, useLiteMode)
```

Comment:
These methods can be replaced with `indexor(etc.).latestApplicableVersion(runNumber)`